ci-operator/step-registry/cucushift/installer/rehearse/azure/upi/custom-nsg/provision/cucushift-installer-rehearse-azure-upi-custom-nsg-provision-chain.yaml

```diff
@@ -0,0 +1,17 @@
+chain:
+  as: cucushift-installer-rehearse-azure-upi-custom-nsg-provision
+  steps:
+  - ref: azure-provision-resourcegroup
+  - ref: azure-provision-vnet
+  - ref: ipi-install-rbac
+  - chain: ipi-conf-azure
+  - ref: ipi-conf-azure-provisioned-resourcegroup
+  - ref: ipi-conf-azure-provisionednetwork
+  - ref: upi-install-azure
```

Comment:
I wonder does this testing only can be executed against UPI clusters?

Reply:
seems only UPI doc has such requirements https://docs.openshift.com/container-platform/4.15/installing/installing_azure/installing-azure-user-infra.html#installation-network-connectivity-user-infra_installing-azure-user-infra , for IPI those should be controlled by the installer.

Reply:
I guess customer vpc + IPI is also applicable for this case.

Reply:
I think it is also applicable for IPI installation in existing vnet, In doc [Installing a cluster on Azure into an existing VNet](https://docs.openshift.com/container-platform/4.15/installing/installing_azure/installing-azure-vnet.html#installation-about-custom-azure-vnet-nsg-requirements_installing-azure-vnet), it mentions nsg requirements, but seems less ports compared with [UPI doc](https://docs.openshift.com/container-platform/4.15/installing/installing_azure/installing-azure-user-infra.html#installation-network-connectivity-user-infra_installing-azure-user-infra), might be doc bug? let me do some test.

Reply:
Following doc [Installing a cluster on Azure into an existing VNet](https://docs.openshift.com/container-platform/4.15/installing/installing_azure/installing-azure-vnet.html#installation-about-custom-azure-vnet-nsg-requirements_installing-azure-vnet) to create nsg for less ports, installation failed. For example, etcd operator is degraded and get error in etcd operator pod.
```
etcd                                       4.16.0-0.nightly-2024-06-18-190130   False       True          True       54m     StaticPodsAvailable: 0 nodes are active; 3 nodes are at revision 0; 0 nodes have achieved new revision 1

W0619 09:00:13.644998       1 etcdcli_pool.go:73] could not create a new cached client after 0 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
I0619 09:00:15.645691       1 etcdcli_pool.go:70] creating a new cached client
W0619 09:00:19.041044       1 etcdcli_pool.go:73] could not create a new cached client after 2 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
E0619 09:00:19.041318       1 base_controller.go:268] ClusterMemberController reconciliation failed: could not get list of unhealthy members: giving up getting a cached client after 3 tries
I0619 09:00:19.041363       1 etcdcli_pool.go:70] creating a new cached client
W0619 09:00:24.888518       1 etcdcli_pool.go:73] could not create a new cached client after 2 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
E0619 09:00:24.888799       1 base_controller.go:268] EtcdEndpointsController reconciliation failed: failed to get member list: giving up getting a cached client after 3 tries
I0619 09:00:24.888902       1 etcdcli_pool.go:70] creating a new cached client
W0619 09:00:25.788043       1 etcdcli_pool.go:73] could not create a new cached client after 2 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
E0619 09:00:25.788088       1 base_controller.go:268] BootstrapTeardownController reconciliation failed: error while canRemoveEtcdBootstrap: giving up getting a cached client after 3 tries
I0619 09:00:25.788324       1 etcdcli_pool.go:70] creating a new cached client
W0619 09:00:30.648176       1 etcdcli_pool.go:73] could not create a new cached client after 1 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
W0619 09:00:34.042695       1 etcdcli_pool.go:73] could not create a new cached client after 0 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
I0619 09:00:34.648540       1 etcdcli_pool.go:70] creating a new cached client
I0619 09:00:36.043818       1 etcdcli_pool.go:70] creating a new cached client
W0619 09:00:39.890771       1 etcdcli_pool.go:73] could not create a new cached client after 0 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
W0619 09:00:40.789792       1 etcdcli_pool.go:73] could not create a new cached client after 0 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
I0619 09:00:41.891798       1 etcdcli_pool.go:70] creating a new cached client
E0619 09:00:41.960164       1 base_controller.go:268] FSyncController reconciliation failed: Post "https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query": dial tcp 172.30.105.209:9091: connect: connection refused
I0619 09:00:42.790742       1 etcdcli_pool.go:70] creating a new cached client
W0619 09:00:49.650074       1 etcdcli_pool.go:73] could not create a new cached client after 2 tries, trying again. Err: failed to make etcd client for endpoints [https://10.0.0.5:2379]: context deadline exceeded
I0619 09:00:49.650257       1 etcdcli_pool.go:70] creating a new cached client
E0619 09:00:49.650351       1 base_controller.go:268] EtcdMembersController reconciliation failed: giving up getting a cached client after 3 tries
```
I think this should be doc bug.
Add azure-ipi-cutom-nsg job to cover custom nsg on ipi + existing vnet.

Reply:
Good job.